crates/turborepo-lib/src/run/mod.rs

```diff
@@ -229,10 +232,37 @@ impl Run {
             self.base.ui,
         ));
 
+        let mut global_env_mode = opts.run_opts.env_mode;
+        if matches!(global_env_mode, EnvMode::Infer)
+            && !root_turbo_json.global_pass_through_env.is_empty()
+        {
+            global_env_mode = EnvMode::Strict;
+        }
+
+        let workspaces = pkg_dep_graph.workspaces().collect();
+        let package_inputs_hashes = PackageInputsHashes::calculate_file_hashes(
+            scm,
+            engine.tasks().par_bridge(),
```

Comment:
Hmm yeah will `concurrency` apply to all forms of threads in our execution? Because we have a few situations like task execution, hashing, caching, etc. that spin up threads.

Reply:
On the Go side it was more of a "best effort" limit to the amount of goroutines we would spin up. In some places we would respect it like with package task hashing, but in other places we wouldn't share that limit. I think we should probably just limit concurrency for stuff where we spend a decent amount of time e.g. hashing, task execution, output saving (?).